### PR TITLE
Feat/utils/slug/non alphanumeric character fix and replace improvement

### PR DESCRIPTION
**Pull request by ilgityildirim**
There was a problem with non-alphanumeric characters as it was turning them into their ASCII counterparts.
**Example String** ````Phalcon's incubator is great````
**Output** ````phalcon39s-incubator-is-great````

Obviously this is not the expected nor wanted output.
**New Output** ````phalcon-s-incubator-is-great````

````$replace```` was also just replacing any given character with a space ```` ```` This might not be the case in some situations such as (with or without using ````$replace````);
**Example String** ````What does it mean yapılır in Turkish````
**Output** ````what-does-it-mean-yaplr-in-turkish````

We can simply pass ```$replace = [ 'ı' => 'i' ]``` with the new update to overcome this problem
**New Output** ````what-does-it-mean-yapilir-in-turkish````

### CODE REVIEW

File: Library/Phalcon/Utils/Slug.php [hunk 1]
```diff
@@ -43,12 +44,20 @@ public static function generate($string, $replace = array(), $delimiter = '-')
         $oldLocale = setlocale(LC_ALL, '0');
         setlocale(LC_ALL, 'en_US.UTF-8');
 
-        $clean = iconv('UTF-8', 'ASCII//TRANSLIT', $string);
+        // replace non letter or non digits by -
+        $string = preg_replace("#[^\\pL\d]+#u", '-', $string);
+
+        // Trim trailing -
+        $string = trim($string,'-');
```
Comment (sergeyklay) [inline Library/Phalcon/Utils/Slug.php]: PSR-2
`No space found after comma in function call`

**Comment by sergeyklay** [comment]
@ianbytchek Ok. PSR fixed. Now unit test fails

**Comment by ilgityildirim** [comment]
@sergeyklay I just installed and executed PHPUnit with it and it failed.
I don't think it has anything to do with the changes I've made. Granted, I've never used PHPUnit before so I might have done something which I over looked.

**Comment by sergeyklay** [comment]
Can you please (for example), remove locally you changes and run test again? ;)

**Comment by ilgityildirim** [comment]
Got the same errors;
* I don't have composer installed
* Tried it with commenting out autoloader.php part as well as I'm just running incubator test without any project.

**Comment by sergeyklay** [comment]
https://github.com/phalcon/incubator/pull/373 latest commit. All ok

**Comment by sergeyklay** [comment]
```sh
20:11:29 [klay@edge:mailfier]$ cd /tmp
20:11:34 [klay@edge:tmp]$ mkdir test
20:11:38 [klay@edge:tmp]$ cd test/
20:11:41 [klay@edge:test]$ php -r "readfile('https://getcomposer.org/installer');" | php
#!/usr/bin/env php
All settings correct for using Composer
Downloading...

Composer successfully installed to: /tmp/test/composer.phar
Use it: php composer.phar
20:11:49 [klay@edge:test]$ ls -al
total 1176
drwxrwxr-x  2 klay klay    4096 Jun 11 20:11 .
drwxrwxrwt 26 root root   90112 Jun 11 20:11 ..
-rwxr-xr-x  1 klay klay 1105755 Jun 11 20:11 composer.phar
```

**Comment by ilgityildirim** [comment]
Installing composer is not a problem although I really prefer not to. Can't I just run unit test without adding it to any project?

**Comment by sergeyklay** [comment]
```sh
$ cd /tmp
$ mkdir test
$ cd test/
$ composer require "phalcon/incubator"
$ ls -al
total 112
drwxrwxr-x  3 klay klay  4096 Jun 11 21:01 .
drwxrwxrwt 26 root root 90112 Jun 11 21:01 ..
-rw-rw-r--  1 klay klay    63 Jun 11 21:01 composer.json
-rw-rw-r--  1 klay klay  4113 Jun 11 21:01 composer.lock
drwxrwxr-x  5 klay klay  4096 Jun 11 21:01 vendor
$ cd vendor/phalcon/incubator/
$ composer install
# ...
# installing dependencies
# ...
$ phpunit --configuration tests/phpunit.xml.dist tests/Phalcon/Utils/SlugTest.php 
PHPUnit 4.6.10 by Sebastian Bergmann and contributors.

Configuration read from /tmp/test/vendor/phalcon/incubator/tests/phpunit.xml.dist

........

Time: 115 ms, Memory: 3.25Mb

OK (8 tests, 10 assertions)
```

**Comment by ilgityildirim** [comment]
@sergeyklay thank you for the clarification and your time. It is crystal clear now how to do unit testing on incubator.

````OK (9 tests, 11 assertions)````

Did the testing before pushing. Once again thank you.

**Comment by sergeyklay** [comment]
You're welcome :)

**Comment by sergeyklay** [comment]
It seems now ok, but something wrong with the travis build. It is no code related issue.  Mb need reopen PR

**Comment by sergeyklay** [comment]
Thanks

**Comment by ilgityildirim** [comment]
@sergeyklay just saw it. Thank you for your time as well. I also got my hands on PHP Unit test :)

**Pull request merged**